### PR TITLE
Delete form

### PR DESCRIPTION
**Pull request by manuelcheveste**
Trello links: 
[As an admin I should be able to delete a form](https://trello.com/c/t0m6PvLB/12-3-as-an-admin-i-should-be-able-to-delete-a-form)
[Forms should have the create form button](https://trello.com/c/6fSyYH7r/13-2-forms-should-have-the-create-form-button)
[Rootstrap should be home link](https://trello.com/c/AkiGQplM/14-1-rootstrap-should-be-home-link)

- [x] Display a list of forms
- [x] Add delete icon to each form item
- [ ] Delete form when the icon is clicked

I added a couple of features which I needed asap, including navigation (desktop and mobile) and transitions for menus.

The next pr will include delete form working as expected.

output

### CODE REVIEW

File: src/components/common/Nav.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      </CSSTransition>
+    </StyledNav>
+  );
+};
+
+Nav.propTypes = {
+  children: PropTypes.node.isRequired
+};
+
+export default Nav;
```
Comment (mcousillas6) [inline src/components/common/Nav.js]: This file is a good example of why I'm not entirely sold on styled 😞 things start looking really messy when transitions & media queries start to appear.

File: src/pages/Forms.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  width: 80%;
+  align-self: center;
+  max-width: 730px;
+`;
+
+const Forms = () => {
+  const formCategories = useSelector(getFormCategories);
+  const forms = useSelector(getFormsByCategory);
+  const error = useSelector(getFormError);
+  const isFetching = useSelector(getFormIsFetching);
```
Comment (mcousillas6) [inline src/pages/Forms.js]: These two should be handled using `const { status, error } = useStatus(FETCH_FORMS)` if you have the `statusReducer` from the react-base

File: src/reducers/formReducer.js [hunk 3]
```diff
@@ -2,12 +2,34 @@ import * as types from 'actions/actionTypes';
...
+  [types.FETCH_FORMS]: state => {
+    state.isFetching = true;
+  },
+  [types.FETCH_FORMS_SUCCESS]: (state, { payload }) => {
+    state.byId[payload.category] = payload.forms;
+    state.isFetching = false;
+  },
+  [types.FETCH_FORMS_ERROR]: (state, { payload }) => {
+    state.error = payload.data.message;
+    state.isFetching = false;
```
Comment (mcousillas6) [inline src/reducers/formReducer.js]: You could omit al this boilerplate if you rely on the statusReducer to manage action's statuses. I really think this https://github.com/rootstrap/react-redux-base/blob/master/src/reducers/statusReducer.js would help a ton (plus the two hooks that are on the hooks folder 😉 )

File: src/components/common/Nav.js (hunk 1 above)
Comment (manuelcheveste) [reply src/components/common/Nav.js]: I'm sorry about that :( Is just that I love so much the power of using JS functions to create css as you can see in this file in line 21

Not only I have the power now of creating html elements using JS thanks to React, now I also can do the same with css thanks to styled components, but yeah, it come with a cost.

Anything I can do to improve code, I'm open for feedback. 
@bernabe9 do you have time this week to review my large files using styled components?

File: src/pages/Forms.js (hunk 2 above)
Comment (manuelcheveste) [reply src/pages/Forms.js]: I will check that

File: src/reducers/formReducer.js (hunk 3 above)
Comment (manuelcheveste) [reply src/reducers/formReducer.js]: Alright, going to take a look to reduce the boilerplate :)

File: src/pages/Forms.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  getFormsByCategory,
+  getFormError,
+  getFormIsFetching
+} from 'selectors/formSelectors';
+import { fetchFormCategories, fetchForms } from 'actions/formActions';
+import Header from 'components/common/Header';
+import Title from 'components/common/Title';
+import Subtitle from 'components/common/Subtitle';
+import Link from 'components/common/Link';
+import Loading from 'components/common/Loading';
```
Comment (xlasserre) [inline src/pages/Forms.js]: Maybe you can export all of these from `common`.

Comment (manuelcheveste) [reply src/pages/Forms.js]: Sounds great, gonna do it

**Review by mcousillas6 — APPROVED** [review]

**Review by xlasserre — APPROVED** [review]

**Pull request merged**